### PR TITLE
Feature/14149037/chanel list

### PR DESCRIPTION
**Pull request by b-dabrowski**

### CODE REVIEW

File: backend/src/common/services/settings.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { SettingsEnum } from "../../controllers/settingsController/models/ISettings"
+import Setting, { ISetting, ISettingDocument } from "../../models/setting.model"
+
+
+
```
Comment (TomaszChmielPagepro) [inline backend/src/common/services/settings.ts]: Remove 2 new lines

File: backend/src/common/services/settings.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+export default class SettingsService {
+  public static getDefaultSettingsIds(settings: ISettingDocument[]) {
+    return settings.map(({ _id }) => _id)
+  }
+
+  private defaultSettings: ISetting[] = [
+    {
+      key: SettingsEnum.BotResponseChannelId,
+      value: ''
```
Comment (TomaszChmielPagepro) [inline backend/src/common/services/settings.ts]: `String.empty`

File: backend/src/common/services/slackClient.ts [hunk 3]
```diff
@@ -53,7 +52,7 @@ export default class SlackClientService {
         if (ok) {
           const { id: generalChannelId } = channels
             .find(({ is_general }) => is_general)
-          SlackClientService.generalChannels[teamId] = generalChannelId
+          SlackClientService.botResponseChannelsIds[teamId] = generalChannelId
```
Comment (TomaszChmielPagepro) [inline backend/src/common/services/slackClient.ts]: New line

File: client-dashboard/src/components/fields/SelectFinal.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { Select } from 'antd'
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/components/fields/SelectFinal.tsx]: `import { Select } from 'antd'` => `import { Select as AntdSelect } from 'antd'`

File: client-dashboard/src/components/fields/SelectFinal.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  label: string
+}
+
+interface IOwnProps {
+  options: IOption[]
+}
+
+type IProps = SelectProps<string> & IOwnProps
+
+const SelectFinal = ({ options, ...selectProps }: IProps): React.SFC<
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/components/fields/SelectFinal.tsx]: `SelectFinal` => `Select`

File: client-dashboard/src/components/settings/SettingsCard.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { Card } from 'antd'
+import React from 'react'
+import styled from 'styled-components'
+
+const CardWrapper = styled.div`
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/components/settings/SettingsCard.tsx]: You can try to do something like this:
```
const CardWrapper = styled(Card)``
```

File: client-dashboard/src/components/settings/SettingsForm.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import SelectFinal, { IOption } from '../fields/SelectFinal'
+import SettingsCard from './SettingsCard'
+
+interface IProps {
+  allChannels: IOption[],
+  botResponseChannelId?: string
+  onSubmit(data: any): void
+}
+
+const SettingsForm = (props: IProps) => (
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/components/settings/SettingsForm.tsx]: Destructure props

File: client-dashboard/src/components/settings/SettingsForm.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        >
+          <Field
+            name="botResponseChannelId"
+            component={SelectFinal({
+              options: props.allChannels
+            })}
+          />
+        </SettingsCard>
+        <Divider />
+        <Button htmlType='submit'>
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/components/settings/SettingsForm.tsx]: I think default is submit

Comment (b-dabrowski) [reply client-dashboard/src/components/settings/SettingsForm.tsx]: without the prop, the button renders like: 
`<button type="button" class="ant-btn" ant-click-animating-without-extra-node="false"><span>Save settings</span></button>`

**Pull request merged**